lib/reflect/reflection.rb

```diff
@@ -27,21 +27,42 @@ def self.build(subject, constant_name, strict: nil, ancestors: nil)
       new(subject, target, strict)
     end
 
-    def call(method_name, arg=nil)
-      unless target.respond_to?(method_name)
+    def call(method_name, ...)
+      unless target_method?(method_name)
         target_name = Reflect.constant(target).name
         raise Reflect::Error, "#{target_name} does not define method #{method_name}"
       end
 
-      arg ||= subject
+      target.public_send(method_name, ...)
+    end
 
-      target.send(method_name, arg)
+    def !(method_name, ...)
+      call(method_name, subject, ...)
     end
 
     def target_accessor?(name, subject=nil)
       subject ||= constant
       subject.respond_to?(name)
     end
+    alias :target_method? :target_accessor?
```

Comment:
target_accessor should be implemented in terms of target_method, not the other way around

Reply:
Also, consider `def target_accessor?` rather than `alias`, since it's a distinct member of Reflection's protocol